eos/utils/integrate-impl.hh

```diff
@@ -114,6 +117,45 @@ namespace eos
             }
         }
     }
+
+    namespace cubature
+    {
+
+        template <size_t dim_>
+        int integrand(unsigned /* ndim */, const double *x, void *data,
+                      unsigned /* fdim */, double *fval)
+        {
+            auto& f = *static_cast<cubature::fdd<dim_> *>(data);
```

Comment:
Since a vector-valued integrand woudl require returning e.g. an ```std::array```, this
 - should be re-labelled as ```scalar_integrand```, and
 - should accept ```fdim``` as a parameter and assert that it is indeed equal to 1,
 - should accept ```ndim``` and assert that it is equal to ```dim_```.

Reply:
I just realize that this is a hot path. Please put the checks in anyway. We can still improve performance with ```#ifdef DEBUG``` later on.

Reply:
If you insist, I put in the checks. But they will be called thousands of times